python/requirements-dev.txt

```diff
@@ -1,6 +1,6 @@
 
 bandit
-numpy<2.0
```

Comment:
Would it make sense to pin the NumPy version to `>=2,<3`?

1. Pinning would make clear that future changes to `dlisio` should follow the new NumPy conventions. 
2. With the name of this PR I would expect that we want NumPy 2.0 now.

Reply:
We should not pin it to version 2 as long as it works with version 1, and we don't have any compelling reason to brak numpy 1 compatibility. This may cause unnecessary dependency conflicts for users that still have a reason to keep using dumpy 1.

When it comes to setting an upper bound for the version either way has its pros and cons. I prefer not to pin unless we discover that we need to.